### PR TITLE
Freemarker native support 

### LINKED ISSUE #223: Freemarker native support

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #223

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions/freemarker/deployment/src/main/java/org/apache/camel/quarkus/component/freemarker/deployment/FreemarkerProcessor.java [hunk 1]
```diff
@@ -16,31 +16,31 @@
...
-    @Record(value = ExecutionTime.RUNTIME_INIT)
-    void warnJvmInNative(JvmOnlyRecorder recorder) {
-        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
-        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    CamelRuntimeBeanBuildItem quteComponent(CamelFreemarkerRecorder recorder, BeanContainerBuildItem beanContainer) {
+        // set the "real" Qute engine to the Camel Qute component
+        return new CamelRuntimeBeanBuildItem(
+                "qute",
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply extensions/freemarker/deployment/src/main/java/org/apache/camel/quarkus/component/freemarker/deployment/FreemarkerProcessor.java:42]: Uh oh?!?!

File: integration-tests/activemq/src/test/java/org/apache/camel/quarkus/component/activemq/it/ActiveMQTest.java [hunk 2]
```diff
@@ -17,14 +17,13 @@
...
 import io.quarkus.test.common.QuarkusTestResource;
-import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.core.Is.is;
 
-@QuarkusTest
+//@QuarkusTest
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply integration-tests/activemq/src/test/java/org/apache/camel/quarkus/component/activemq/it/ActiveMQTest.java:26]: Nope, thanks for catching it!

**Comment by ppalaga** [comment]
7feca462912f079b6532a5701c38ed64b82fac2f: fixed the two comments by @jamesnetherton

**Review by jamesnetherton — APPROVED** [review]
Sorry to nitpick. The method name is still `quteComponent`. Fix up in a future commit if you don't want to burn CI cycles. Otherwise, LGTM.

**Comment by ppalaga** [comment]
> Sorry to nitpick. The method name is still `quteComponent`. Fix up in a future commit if you don't want to burn CI cycles. Otherwise, LGTM.

So sorry to steel your time. Fixed in a534277 Thanks again!

**Review by aldettinger — APPROVED** [review]
The failing test is related. Maybe the native resources inclusion pattern does not embed sub-folders. Otherwise, looks good to me :+1:

**Comment by ppalaga** [comment]
The test is failing because I forgot that a couple of weeks ago, I hacked around some limitations in Quarkiverse Freemarker and I need to get them merged there first. I am getting old :-/

**Comment by ppalaga** [comment]
Waiting for freemarker-quarkus 0.2.0

**Comment by ppalaga** [comment]
Quarkiverse Freemarker 0.2.0 is available on the Central

**Comment by ppalaga** [comment]
2e637a8: upgraded to Quarkiverse Freemarker 0.2.2

**Pull request merged**